src/explorer/collectionTreeItem.ts

```diff
@@ -44,79 +57,70 @@ export default class CollectionTreeItem extends vscode.TreeItem implements TreeI
     if (this.isExpanded) {
       if (this._childrenCacheIsUpToDate) {
         return Promise.resolve(this._childrenCache);
-      } else {
-        return new Promise((resolve, reject) => {
-          const namespace = `${this._databaseName}.${this._collectionName}`;
-
-          this._dataService.find(
-            namespace,
-            { /* No filter */ },
-            {
-              // We fetch 1 more than the max documents to show to see if
-              // there are more documents we aren't showing.
-              limit: 1 + this._maxDocumentsToShow
-            },
-            (err: any, documents: any[]) => {
-              if (err) {
-                return reject(`Unable to list documents: ${err}`);
-              }
-
-              this._childrenCacheIsUpToDate = true;
-
-              if (documents) {
-                this._childrenCache = documents.map(
-                  (document, index) => index === this._maxDocumentsToShow ?
-                    new ShowMoreDocumentsTreeItem(namespace, this.onShowMoreClicked, this._maxDocumentsToShow)
-                    : new DocumentTreeItem(document)
-                );
-              } else {
-                this._childrenCache = [];
-              }
+      }
 
-              return resolve(this._childrenCache);
+      return new Promise((resolve, reject) => {
+        const namespace = `${this._databaseName}.${this._collectionName}`;
+
+        this._dataService.find(
+          namespace,
+          {
+            /* No filter */
+          },
+          {
+            // We fetch 1 more than the max documents to show to see if
+            // there are more documents we aren't showing.
+            limit: 1 + this._maxDocumentsToShow
+          },
+          (err: any, documents: any[]) => {
+            if (err) {
+              return reject(`Unable to list documents: ${err}`);
             }
-          );
-        });
-      }
+
+            this._childrenCacheIsUpToDate = true;
+
+            if (documents) {
+              this._childrenCache = documents.map((document, index) => {
+                const showMore = new ShowMoreDocumentsTreeItem(
+                  namespace,
+                  this.onShowMoreClicked,
+                  this._maxDocumentsToShow
+                );
+                return index === this._maxDocumentsToShow
+                  ? showMore
```

Comment:
I haven't found a combination of rules to get rid of the indent error, therefore I refactored this block.

https://eslint.org/docs/rules/